Poul12/SimpleToDoApp/src/main/java/pl/_1024kb/stowarzyszenienaukijavy/simpletodo/model/service/UserServiceImpl.java

```diff
@@ -0,0 +1,194 @@
+package pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.api.UserDao;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.api.UserService;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.daojdbc.DaoFactory;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.entity.User;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.exception.*;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.utility.MD5Hash;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+
+public class UserServiceImpl implements UserService
+{
+    private static UserServiceImpl instance;
+    private DaoFactory factory = DaoFactory.getDaoFactory(DaoFactory.MYSQL_DAO);
+    private UserDao dao = factory.getUserDao();
+    //private MysqlUserDao jdbcDao = MysqlUserDao.getInstance();
+    private UserValidator validator = UserValidator.getInstance();
+    private static final Logger logger = LoggerFactory.getLogger(TaskServiceImpl.class);
+
+    private UserServiceImpl()
+    {
+        if(instance != null)
+            throw new IllegalStateException("Cannot create new instance, use getInstance");
+    }
+
+    public static UserServiceImpl getInstance()
+    {
+        if(instance == null)
+            instance = new UserServiceImpl();
+
+        return instance;
+    }
+
+    @Override
+    public String createUser(User user)
+    {
+        String messageInfo = "Pomyślnie udało się zapisać użytkownika do bazy :)";
+        try
+        {
+            if(validator.isUserValid(user) && !isUsernameAlreadyExist(user.getUsername()))
+            {
+                try
+                {
+                    user.setPassword(encodePassword(user));
+                    dao.create(user);
+                    logger.info(messageInfo + " - " + user.getUsername());
+
+                }catch(SQLException e)
+                {
+                    e.printStackTrace();
+                    String messageError = "Nie udało się zapisać użytkownika do bazy";
+                    logger.error(messageError + " - " + user.getUsername());
+                    return messageError;
+                }
+            }
+
+        }catch(TooShortUsernameLengthException | TooShortPasswordLengthException | NotValidUserEmailException | NotTheSamePasswordException e)
```

Comment:
Lepiej zrobić te wyjątki unchecked - wyjątki checked powinny być stosowane wtedy, gdy mamy naprawdę poważny błąd lub integrujemy się z cudzym serwisem. Tutaj mamy wyjątki z walidatora. Możesz złapać tutaj Exception - zrobić logowanie akcji i rzucić wyjątek dalej, aby servlet zajął się wyświetleniem informacji. Ewentualnie zwrócić to przez Stringa - choć zazwyczaj w tym celu stosuje się wyjątki bo w końcu służą obsłudze błędów :P

Reply:
Poprawiłem obsługę wyjątków w serwisie, łapie Exception i rzucam do servletów i tam podaje informacje które wyświetlam na stronie.